### PR TITLE
New email verification after previous expired

### PR DESCRIPTION
**Pull request by emmacode**
[Issue 416](https://github.com/boxyhq/saas-starter-kit/issues/416)

- A new verification link gets sent to the user immediately when they click on the expired verification link. 

Please if there is a space for improvement. I will appreciate the feedback @devkiran

### CODE REVIEW

**Comment by devkiran** [comment]
@emmacode Let me discuss this internally and get back to you.

**Comment by emmacode** [comment]
Thank you!

**Comment by emmacode** [comment]
@devkiran any feedback on this yet?

**Comment by devkiran** [comment]
Here are more details about how we want this feature.

If the email verification link is expired, display a form where they can enter their email address and request a new verification email. Make sure we validate the email exists before sending the email.

The form can be a simple form with an input and button similar to the request password form. 

Title: Verify your account
Description: This verification link is expired. Please enter the email address associated with your account, and we'll send you another verification link.

After sending the verification email, display a toast and take them to the logic screen (Maybe?)

**Comment by emmacode** [comment]
Alright will do that

**Comment by emmacode** [comment]
@devkiran feature has been implemented as described. Will appreciate feedback

File: pages/api/auth/verify-account.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { generateToken, validateEmail } from '@/lib/common';
```
Comment (devkiran) [inline pages/api/auth/verify-account.ts:1]: Can we rename this API to `/resend-email-token`?

File: pages/auth/verify-account.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  const formik = useFormik({
+    initialValues: {
+      email: '',
+    },
+    validationSchema: Yup.object().shape({
+      email: Yup.string().required().email(),
+    }),
+    onSubmit: async (values) => {
+      const response = await fetch('/api/auth/verify-account', {
```
Comment (devkiran) [inline pages/auth/verify-account.tsx:45]: Please update this endpoint to `/api/auth/resend-email-token`.

File: pages/auth/verify-account.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      const json = (await response.json()) as ApiResponse;
+
+      if (!response.ok) {
+        toast.error(json.error.message);
+        return;
+      }
+
+      formik.resetForm();
+      toast.success(t('verify-account-link-sent'));
+      router.push('/login');
```
Comment (devkiran) [inline pages/auth/verify-account.tsx:60]: After sending the verification email, let's redirect the users to `/auth/verify-email`.

File: pages/auth/verify-account.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import { AuthLayout } from '@/components/layouts';
```
Comment (devkiran) [inline pages/auth/verify-account.tsx:1]: Can we rename this page to `/auth/resend-email-token`?

**Review by devkiran — CHANGES_REQUESTED** [review]
LGTM. Added some comments to update the routes.

**Comment by emmacode** [comment]
Alright!

File: pages/api/auth/verify-account.ts (hunk 1 above)
Comment (emmacode) [reply pages/api/auth/verify-account.ts:1]: done

**Comment by emmacode** [comment]
@devkiran. I appreciate the feedback. All requested changes as been done

**Review by devkiran — APPROVED** [review]

**Comment by NathanTarbert** [comment]
Amazing @emmacode! 

Thanks so much for the wonderful contribution :)

**Pull request merged**